### ISSUE TITLE
Server 500 error when viewing a bbb-playback-screenshare recording

### ISSUE THREAD

**Issue by ffdixon** [bigbluebutton/greenlight#845]
On a clean install with the latest version of Greenlight, it correctly displays a presentation recording with screen share.

![image](https://user-images.githubusercontent.com/290351/67153826-6264ec00-f2be-11e9-87da-aea2a5c06e71.png)

However, after adding the `bbb-playback-screenshare` package and rebuilding the recording, Greenlight gives a 500 error.

![image](https://user-images.githubusercontent.com/290351/67153831-80325100-f2be-11e9-8996-69ad655dac14.png)

Looking at the XML returned back from `getRecordings` API call,

~~~xml
<?xml version="1.0" encoding="UTF-8"?>
<response>
   <returncode>SUCCESS</returncode>
   <recordings>
      <recording>
         <recordID>2d86ce91c5039fd9bc54d04f0b49121d12908d0a-1571537136631</recordID>
         <meetingID>2d86ce91c5039fd9bc54d04f0b49121d12908d0a-1571537136631</meetingID>
         <name>Home Room</name>
         <isBreakout>false</isBreakout>
         <published>true</published>
         <state>available</state>
         <startTime>1571537136631</startTime>
         <endTime>1571537168058</endTime>
         <participants>0</participants>
         <rawSize>0</rawSize>
         <metadata>
            <bbb-origin-version>2.4-b2</bbb-origin-version>
            <meetingName>Home Room</meetingName>
            <meetingId>66fef7abb45cb49dff6477925c19ba71c5649879</meetingId>
            <gl-listed>false</gl-listed>
            <bbb-origin>Greenlight</bbb-origin>
            <isBreakout>false</isBreakout>
            <bbb-origin-server-name>droplet-5026.example.com</bbb-origin-server-name>
         </metadata>
         <size>1482416</size>
         <playback>
            <format>
               <type>screenshare</type>
               <url>https://droplet-5026.example.com/recording/screenshare/2d86ce91c5039fd9bc54d04f0b49121d12908d0a-1571537136631/</url>
               <processingTime>0</processingTime>
               <length>0</length>
               <size>0</size>
            </format>
            <format>
               <type>presentation</type>
               <url>https://droplet-5026.example.com/playback/presentation/2.0/playback.html?meetingId=2d86ce91c5039fd9bc54d04f0b49121d12908d0a-1571537136631</url>
               <processingTime>17947</processingTime>
               <length>0</length>
               <size>1482416</size>
               <preview>
                  <images>
                     <image alt="Welcome To BigBlueButton" height="136" width="176">https://droplet-5026.example.com/presentation/2d86ce91c5039fd9bc54d04f0b49121d12908d0a-1571537136631/presentation/d2d9a672040fbde2a47a10bf6c37b6a4b5ae187f-1571537137803/thumbnails/thumb-1.png</image>
                     <image alt="This slide left blank for whiteboard" height="136" width="176">https://droplet-5026.example.com/presentation/2d86ce91c5039fd9bc54d04f0b49121d12908d0a-1571537136631/presentation/d2d9a672040fbde2a47a10bf6c37b6a4b5ae187f-1571537137803/thumbnails/thumb-2.png</image>
                     <image alt="This slide left blank for whiteboard" height="136" width="176">https://droplet-5026.example.com/presentation/2d86ce91c5039fd9bc54d04f0b49121d12908d0a-1571537136631/presentation/d2d9a672040fbde2a47a10bf6c37b6a4b5ae187f-1571537137803/thumbnails/thumb-3.png</image>
                  </images>
               </preview>
            </format>
         </playback>
         <data />
      </recording>
   </recordings>
</response>
~~~

Here we see the XML returned for `screenshare` format is minimal

~~~xml
            <format>
               <type>screenshare</type>
               <url>https://droplet-5026.example.com/recording/screenshare/2d86ce91c5039fd9bc54d04f0b49121d12908d0a-1571537136631/</url>
               <processingTime>0</processingTime>
               <length>0</length>
               <size>0</size>
            </format>
~~~

While we could update Greenlight to be more forgiving of parsing the XML, it might be better to update `bbb-playback-screenshare` to be consistent with other formats (to avoid breaking other front-ends).

Some more discussion is need on this on the best approach, but at the end we want the `bbb-playback-screenshare` to work in BigBlueButton 2.2 with Greenlight

**Comment by farhatahmad** [comment]
> While we could update Greenlight to be more forgiving of parsing the XML, it might be better to update bbb-playback-screenshare to be consistent with other formats

I think the solution here would be to do both of these. While all recording formats should be consistent, a bad recording format should not make that room in Greenlight fully unusable. It would be better to display some sort of error in the recording table instead.

**Comment by farhatahmad** [comment]
The issue appears to be here:
```<recordID>2d86ce91c5039fd9bc54d04f0b49121d12908d0a-1571537136631</recordID>
 <meetingID>2d86ce91c5039fd9bc54d04f0b49121d12908d0a-1571537136631</meetingID>```

Which implies that it's an issue with the response to getMeetings.

PR #849 changes how errors with recordings are handled, meaning that any errors that happen when displaying a recording will result in an alert informing the user of the number of recordings in which there was errors, while still allowing the room to be accessible.

**Comment by mhajim** [comment]
Hi,
I have installed new BBB (2.4) and move my greenlight backup(DB+recorded files) from old BBB(2.2) and it works but the problem is recorded videos!!

 I got this error:
500 Internal Server Error

URL: https://my-domain.com/playback/presentation/2.3/b62490c260a8eac2d9df5b8b68f20db0828616c7-1641002799791
Old Greenlight. release-2.9.2
New Greenlight. release-2.10.0.3

/var/log/nginx/error.log:
*6415 rewrite or internal redirection cycle while internally redirecting to "/playback/presentation/2.3/index.html", client: 1.1.1.1!, server: my-domain.com, request: "GET /playback/presentation/2.3/b62490c260a8eac2d9df5b8b68f20db0828616c7-1641002799791 HTTP/2.0", host: "lms1.ystp.ac.ir", referrer: "https://my-domain.com/b/admins/recordings"

I run bbb-record --rebuildall but still have an error!
